core/kernel/classes/class.Class.php

```diff
@@ -228,6 +228,52 @@ public function createSubClass($label = '', $comment = '', $uri = "")
         return $returnValue;
     }
 
+    /**
+     * Retrieves a direct subclass by label.
+     *
+     * @param  string label
+     * @return core_kernel_classes_Class|null
+     */
+    public function retrieveSubClassByLabel($label)
+    {
+        $subClasses = $this->getSubClasses();
+        foreach ($subClasses as $subclass) {
+            if ($subclass->getLabel() === $label) {
+                return $subclass;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Retrieves a direct subclass by label or creates it if not existent.
+     *
+     * @param  string label
+     * @return core_kernel_classes_Class
+     */
+    public function retrieveOrCreateSubClassByLabel($label)
+    {
+        return $this->retrieveSubClassByLabel($label) ?: $this->createSubClass($label);
+    }
+
+    /**
+     * Creates a path of subclasses from an array of labels, URIs and comments.
+     *
+     * @param  array $labels indexed array of labels ordered from root to leaf class
+     * @return core_kernel_classes_Class The last class created
+     */
+    public function createSubClassPathByLabel(array $labels)
+    {
+        $currentClass = $this;
+
+        foreach($labels as $label) {
+            $currentClass = $currentClass->retrieveOrCreateSubClassByLabel($label);
+        }
```

Comment:
I don't think we need to go through all array `$labels`
seems it is enough to `retrieveOrCreateSubClassByLabel` only the last item in the array

Reply:
Not really, `createClass`, if it occurs along the path, triggers `ResourceCreated`. So all sorts of side effects can happen in its handlers.

Reply:
We have to do it for each level of the tree, since some (at the beginning) can already exist. We then retrieve the existing ones and create the non-existing ones.
`retrieveOrCreateSubClassByLabel` only creates one subclass.